### PR TITLE
[Release/8.0-staging] Reduce net core app current package dependencies

### PR DESCRIPTION
**Pull request by ericstj**
## Customer Impact

- [x] Customer reported
- [x] Found internally

Customers using packages like `Microsoft.Extensions.Logging.Console`, `System.Memory.Data`, `Microsoft.Extensions.Hosting`, etc and targeting `net8.0` need to update package references for packages like `System.Text.Json`, `System.Text.Encodings.Web`, `System.Collections.Immutable`, etc even though those libraries are provided by the shared framework.

We can reduce the number of places where folks need to update packages by omitting packages when the same library is provided by the framework.  In particular this will fix Aspire templates so that they'll not need to add packages in servicing to avoid vulnerability warnings from NuGet audit.

## Regression

- [ ] Yes
- [X] No

## Testing

Build packages.  Add validation to make sure all up-stack packages ship as well.  Tested these updates with Aspire template to confirm that NuGet audit warnings go away - 2 framework packages were dropped (JSON and STEW).

## Risk

Low.  This is removing package references for a few packages and enabling more packages.  The biggest risk here is that we'll be enabling a lot of packages that need to flow in servicing.

## Background

This drops package dependencies from all packages which can reference the framework copy of the same library (without downgrading the library exposed to a compatible TFM like netstandard2.0).

I also added a feature to our build that enforces transitive servicing.  You can specify `ServiceTransitiveDependencies` to make sure that you enable all up-stack packages for shipping when enabling a single package.

I enabled all packages I changed, then all upstack packages (separate commits).

This should improve the situation where folks are asked to update just to update a package reference on the latest framework.

It will also help reduce application size since the libraries will no longer be bundled in the app.

### CODE REVIEW

**Comment by ericstj** [comment]
This is a backport of the fix for https://github.com/dotnet/runtime/issues/105120

File: eng/packaging.targets
Comment (eerhardt) [inline eng/packaging.targets:358]: Does this need to be ported to `main`? I didn't see it in the original PR - https://github.com/dotnet/runtime/pull/106172/

Comment (ericstj) [reply eng/packaging.targets:358]: This is for servicing to help enabling packages when servicing dependencies.  We may not need it in `main` if we decide to stop doing incremental servicing.

Comment (eerhardt) [reply eng/packaging.targets:358]: > if we decide to stop doing incremental servicing

I didn't know that was an option on the table.I honestly don't know what I think about it. There are pros and cons either way, for sure.

File: docs/project/library-servicing.md
Comment (steveharter) [inline docs/project/library-servicing.md:21]: typo: "with" -> "wish".

Who are the expected users of `ServiceTransitiveDependencies=true` ? Assuming incremental servicing, wouldn't we want to use this for our own OOB packages?

Comment (ericstj) [reply docs/project/library-servicing.md:21]: The expected users would be those building the entire repo and wanting all up-stack packages to be updated.   It's currently not the default behavior, but it's useful in this PR since to see the impact of removing dependencies we need to update all packages that will be impacted by those removals.

I might refine this to also allow it to be specified optionally in the serviced package too.  That would let folks do this more granularly if they wanted.  For now, this is primarily an off-by-default experiment to see if we might wish to use this workflow in the future.

**Review by carlossanlop — APPROVED** [review]
LGTM after your explanation in our call.

**Comment by carlossanlop** [comment]
/ba-g The unknown failure seems unrelated to this PR. I opened https://github.com/dotnet/runtime/issues/107413 to track it.

**Comment by carlossanlop** [comment]
@ericstj do you want to do the honors and smash the squash and merge button?

**Pull request merged**
